### PR TITLE
fix kube discovery to work with named ports

### PR DESCRIPTION
**Pull request by ilackarms**
this moves the burden of discovering the kube service's `TargetPort` to `EDS`. formerly UDS was looking up target port directly on the service; however this doesn't support the case where target port is named on the kubernetes `Endpoint`.

### CODE REVIEW

File: projects/gloo/pkg/plugins/kubernetes/eds.go [hunk 1]
```diff
@@ -124,13 +150,14 @@ func filterEndpoints(ctx context.Context, writeNamespace string, kubeEndpoints [
 			for _, subset := range eps.Subsets {
 				var port uint32
 				for _, p := range subset.Ports {
-					if spec.TargetPort == uint32(p.Port) {
+					switch {
+					case p.Name == kubeServicePort.Name:
 						port = uint32(p.Port)
 						break
 					}
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/kubernetes/eds.go]: i think you forgot to add the case of pod port is not named, and service has a target port.

Comment (ilackarms) [reply projects/gloo/pkg/plugins/kubernetes/eds.go]: if the port pod isn't named, it means [there can only be one](https://prblog.typepad.com/.a/6a00d8341c57a853ef0120a4d0501a970b-500wi) in which case we should ignore the target port altogether

Comment (yuval-k) [reply projects/gloo/pkg/plugins/kubernetes/eds.go]: yes, but in this case the only option that works is a name port.
or are you relying on the fact that they both will have an empty string for a name?
if that's the case, please add a comment to clarify

Comment (ilackarms) [reply projects/gloo/pkg/plugins/kubernetes/eds.go]: will add comment

**Pull request merged**